### PR TITLE
feat(progressring): WinUI 2.4 ProgressRing

### PR DESCRIPTION
**Pull request by carldebilly**
# Feature
Implemented `Microsoft.Xaml.UI.Controls.ProgressRing` for all platforms using **Lottie Animations**!

To use it, the package `Uno.UI.Lottie` is required in the app or nothing will be displayed by default.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [X] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressRing/ProgressRing.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		}
+
+		private void ChangeVisualState()
+		{
+			if (IsActive)
+			{
+				// Support for older templates
+				VisualStateManager.GoToState(this, "Active", true);
+
+				var _ = _player?.PlayAsync(0, 1, true);
```
[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/Microsoft/UI/Xaml/Controls/ProgressRing/ProgressRing.cs:92]: You can probably just use `_` instead.

File: src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs [hunk 2]
```diff
@@ -133,22 +146,75 @@ Size IAnimatedVisualSource.Measure(Size availableSize)
...
+					this.Log().Warn($"Unable to find embedded resource named '{resourceName}' to load.");
+				}
+				json = null;
+				return false;
 			}
 
-			return availableSize;
+			var bytes = new byte[(int)stream.Length];
+			stream.Read(bytes, 0, bytes.Length);
+			json = Encoding.UTF8.GetString(bytes);
```
Comment (jeromelaban) [inline src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs:216]: May not be UTF8, `StreamReader` may be more appropriate here ?

File: src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs [hunk 3]
```diff
@@ -133,22 +146,75 @@ Size IAnimatedVisualSource.Measure(Size availableSize)
...
+			if (assemblyName == ".")
+			{
+				assembly = Application.Current.GetType().Assembly;
+			}
+			else
+			{
+				assembly = AppDomain
+					.CurrentDomain.GetAssemblies()
+					.FirstOrDefault((Assembly a) =>
+						a.GetName().Name.Equals(assemblyName, StringComparison.OrdinalIgnoreCase));
```
Comment (jeromelaban) [inline src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs:193]: The use of `Assembly.Load(assemblyName)` may be faster than scanning all domain assemblies, as runtimes do that completely natively.

File: src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs (hunk 2 above)
Comment (carldebilly) [reply src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs:216]: Lottie Files are all exported from a tool and always UTF-8.

File: src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs (hunk 3 above)
Comment (carldebilly) [reply src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs:193]: I'll try that.

Comment (carldebilly) [reply src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs]: Done

**Pull request merged**